dashboard/app/models/pd/district_report.rb

```diff
@@ -0,0 +1,56 @@
+class Pd::DistrictReport
+
+  # Construct a report row for each teacher in the district
+  def self.generate_district_report(*districts)
+    [].tap do |rows|
+      districts.each do |district|
+        district.users.all.each do |teacher|
+          Pd::Workshop.attended_by(teacher).each do |workshop|
+            rows << generate_district_report_row(district, teacher, workshop)
+          end
+        end
+      end
+    end
+  end
+
+  def self.generate_district_report_row(district, teacher, workshop)
+    payment_term = Pd::DistrictPaymentTerm.where(district_id: district.id, course: workshop.course).first
+    attendances = Pd::Attendance.for_teacher_in_workshop(teacher, workshop)
+    hours = attendances.map(&:hours).reduce(&:+)
+    days = attendances.count
+    qualified = (payment_term && workshop.workshop_type == Pd::Workshop::TYPE_DISTRICT && workshop.course != Pd::Workshop::COURSE_CSF)
+    payment_type = payment_term ? payment_term.rate_type : nil
+    payment_rate = payment_term ? payment_term.rate : nil
+    payment_amount = !qualified ? 0 :
+      case payment_term.rate_type
+        when 'hourly'
+          hours * payment_term.rate
+        when 'daily'
+          days * payment_term.rate
+        else
+          0
```

Comment:
Are hours and daily the only possibilities here? If so, then raise an exception if you get here, or make this statement if else